src/backend/common/legacy_protobuf/legacy_gae_entity_pb_encoder.py

```diff
@@ -369,7 +369,13 @@ def _byte_size_property_value(self: PropertyValue) -> int:
         if self.has_booleanvalue_:
             n += 2
         if self.has_stringvalue_:
-            n += 1 + EntityProtoByteSize.lengthString(len(self.stringvalue_))
+            assert isinstance(self.stringvalue_, (str, bytes))
+            if isinstance(self.stringvalue_, str):
+                str_len = len(cast(str, self.stringvalue_).encode())
```

Comment:
this bit is the key bug fix, since this is what we also do when we write the string